infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/Utils/CorfuSaasEndpointProvider.java

```diff
@@ -0,0 +1,57 @@
+package org.corfudb.infrastructure.logreplication.infrastructure.Utils;
+
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Optional;
+import java.util.Properties;
+
+/**
+ * Helper methods to retrieve the corfu's SaaS endpoint.
+ */
+@Slf4j
+public class CorfuSaasEndpointProvider {
+    private static Optional<CorfuSaasEndpointProvider> instance = Optional.empty();
+
+    @Getter
+    private final String configFile;
+
+    @Getter
+    private final Optional<String> corfuSaasEndpoint;
+
+    private CorfuSaasEndpointProvider(String configFile) {
+        this.configFile = configFile;
+        this.corfuSaasEndpoint = getEndpoint(configFile);
+    }
+
+    public static void init(String pluginConfigFile) {
+        instance = Optional.of(new CorfuSaasEndpointProvider(pluginConfigFile));
+    }
+
+    private Optional<String> getEndpoint(String pluginConfigFilePath) {
+        if (pluginConfigFilePath != null) {
+            return extractEndpoint(pluginConfigFilePath);
+        }
+        log.warn("No plugin path found");
+        return Optional.empty();
+    }
+
+    private Optional<String> extractEndpoint(String path) {
```

Comment:
since this method specifically looks for the saas endpoint, can it be named accordingly?  - extractSaasEndpoint() or something similar?